### DISCUSSION TITLE
Is possible that we can fix timing of subtitles using whisper models

### DISCUSSION THREAD

**Discussion by starbez** [SubtitleEdit/subtitleedit#6399] [General]
Is it possible that using Whisper AI model we only sync/fix subs timing to audio rather than generating them from scratch?

**Comment by niksedk** [comment]
That would work very poorly as Whisper is really bad at finding timestamps.

**Comment by Baenwort** [comment]
There is a project working on improving the timestamps: https://github.com/jianfch/stable-ts perhaps their work could replace the python whisper you use?

**Reply by niksedk** [reply]
SE uses similar stuff, but it only works when there is silence between speech (waiting for this to be fixed in Whisper).

**Comment by Baenwort** [comment]
There is another project that is trying to improve timings and seems to work according to initial reviews: https://github.com/openai/whisper/discussions/435#discussioncomment-4405278 references https://github.com/m-bain/whisperX

**Reply by dgoryeo** [reply]
@Baenwort , just a note: WhisperX at the moment supports very limited number of languages: EN, DE, IT, FR. Hopefully the community finds a way to add more languages to that appoach.

**Reply by dgoryeo** [reply]
Just an update to my earlier message: WhisperX now supports additional non-European languages.

**Reply by chenlung** [reply]
Something was recent committed from WhisperX to Subtitle Edit (partial support? Not sure if it's available/enabled in builds?), so it would be interesting to read your thoughts, both on this issue and audio-to-text in general (barriers; what might be coming, etc.), @niksedk.
